acme/challenge.go

```diff
@@ -175,7 +184,7 @@ func (bc *baseChallenge) getError() *AError {
 
 // toACME converts the internal Challenge type into the public acmeChallenge
 // type for presentation in the ACME protocol.
-func (bc *baseChallenge) toACME(db nosql.DB, dir *directory, p provisioner.Interface) (*Challenge, error) {
```

Comment:
every `toACME` method has the same interface. We can change that, but maybe that should be part of a different PR. I don't really, other than it's nice having all those methods having the same API. Even if one arg isn't used by some of the methods.

Reply:
Perhaps this would be useful additional information to include in a comment and or to codify via an explicit interface that all the acme data types adopt. I removed the the parameter because it's unused, as you mentioned.

Reply:
Eh, you're probably right. In my mind it was more like "if we're gonna change this we may as well change all the other instances too". We can change them over time though.

Reply:
I can make an interface if we want. Up to you!

Reply:
I don't think it's worth it. I thought there would be some benefit to having those methods all have the same function parameters, but I don't think there was any.